### ISSUE TITLE
F10(step over) end of main function, 'crtexe.c' The editor could not be opened because the file was not found.

### ISSUE THREAD

**Issue by Raduq91** [WebFreak001/code-debug#412]
How to fix this issue?

GNU gdb (GDB) 10.2; TDM-GCC-64 version 10.3.0; vscodium 1.85.2; win 10 x64

![er](https://github.com/WebFreak001/code-debug/assets/31180871/33b1b579-990e-4fce-bee1-664f917d0757)

**Comment by GitMensch** [comment]
There's nothing to fix here - you step out of main and therefore into the crt. `skip file` won't help you here as this is only about not stepping _into_ something, and you _may_ even want to debug crtexe.c.

If you want to get rid of this (not using any stepping at the end of main would be easier) then you can either have that C file (actually it could be an empty one with a plain comment "don't step here") or possibly [strip debug information from the MinGW runtime](https://github.com/msys2/MINGW-packages/issues/6653) or use an auto command to create a silent breakpoint at the return address of your main, then do a `continue` there...

Feel free to note something here which is useful in your case, but even than - this is MinGW specific and not related to this extension.

**Comment by eulerbaby123** [comment]
so, if have any method to solve this problem at once?

**Comment by Raduq91** [comment]
> so, if have any method to solve this problem permanently?


I switched a long time ago to msvc and CodeLLDB (vadimcn). I changed some settings to the extension, I don't remember if it had no issues before the changed settings.
In CodeLLDB settings checked the option `Supress VSCode's messages about missing source files (when debug info refers to files not available on the local machine)` and added some init commands(found info in the internet and chatgpt help I think):
![image](https://github.com/user-attachments/assets/29bba7f2-3250-45d2-95da-8467c2071d0b)

No issues at printf, when reached the end of the main function I have to press F10(step over)4 times before debugger exits.

**Comment by Raduq91** [comment]
> so, if have any method to solve this problem permanently?

I approached the problem again(I used msvc), I used CodeLLDB and in the extension settings InitCommands section I added this for printf issue, it does not enter into it:
target stop-hook add --one-liner "script if lldb.frame.GetLineEntry().GetFileSpec().GetFilename()=='stdio.h' : lldb.debugger.HandleCommand('thread step-out')"

And same for issue with entering into exe_common.inl after main returns:
target stop-hook add --one-liner "script if lldb.frame.GetLineEntry().GetFileSpec().GetFilename()=='exe_common.inl' : lldb.debugger.HandleCommand('thread step-out')"

Other mention:
No printf issue with VScodium Version: 1.95.2 and MSVC_VERSION=14.41.34120 (I think maybe it is because of newer msvc)

---

Tested TDM GCC with lldb, modified command above with crtexe.c instead of stdio.h avoid stepping into printf.  LLDB step does not work correctly.

---

Tested LLVM-18.1.8-win64 with LLDB, exe_common.inl issue as above, no printf issue, stepping works correctly. Local variables from left panel disappear when in twoSum for loop.

---

```
LLVM_MINGW64\bin>c++ --version
clang version 18.1.1 (https://github.com/llvm/llvm-project.git dba2a75e9c7ef81fe84774ba5eee5e67e01d801a)
Target: x86_64-w64-windows-gnu
Thread model: posix
(I don't remember exactly, probably I got this from https://github.com/mstorsjo/llvm-mingw)

built with > c++ -g  test.cpp -lstdc++ -o test
```
LLDB steps correctly, Local variables from left panel are visible in for loop, no printf issue, it enters crtexe.c after main return workaround with:
lldb.frame.GetLineEntry().GetFileSpec().GetFilename()=='crtexe.c' : lldb.debugger.HandleCommand('thread step-out')"
